core/src/main/java/com/rudderstack/android/sdk/core/RudderServerConfigManager.java

```diff
@@ -157,8 +160,10 @@ private void sleep(int retryCount) {
         } catch (InterruptedException ex) {
             ReportManager.reportError(ex);
             RudderLogger.logError(String.format(Locale.US, "RudderServerConfigManager: Sleep: Exception while the thread is in sleep %s", ex.getLocalizedMessage()));
+            Thread.currentThread().interrupt();
```

Comment:
Why have we added this?

Reply:
this is added to fix the sonarcloud issues. more on it [here](https://rules.sonarsource.com/java/tag/error-handling/RSPEC-2142/)